internal/graph/graph.go

```diff
@@ -94,7 +94,7 @@ func (g *Graph) InsertConstructor(ctor interface{}) error {
 		objTypes[i] = ctype.Out(i)
 	}
 
-	if err := g.ValidateReturnTypes(ctype); err != nil {
+	if err := g.ValidateReturnTypes(ctype, false); err != nil {
```

Comment:
for Invoke we only want to check if object is cached or not, and ignore provided unresolved constructor.